jenkins_jobs/modules/publishers.py

```diff
@@ -2456,37 +2456,117 @@ def base_email_ext(registry, xml_parent, data, ttype):
+                            email, "recipientProviders"
+                        )
+                    XML.SubElement(
+                        recipient_providers,
+                        "hudson.plugins.emailext.plugins.recipients.RequesterRecipientProvider",
+                    ).text = ""
+                elif "culprits" == recipient:
+                    if recipient_providers is None:
+                        recipient_providers = XML.SubElement(
+                            email, "recipientProviders"
+                        )
+                    XML.SubElement(
+                        recipient_providers,
+                        "hudson.plugins.emailext.plugins.recipients.CulpritsRecipientProvider",
+                    ).text = ""
+                elif "recipients" == recipient:
+                    if recipient_providers is None:
+                        recipient_providers = XML.SubElement(
+                            email, "recipientProviders"
+                        )
+                    XML.SubElement(
+                        recipient_providers,
+                        "hudson.plugins.emailext.plugins.recipients.ListRecipientProvider",
+                    ).text = ""
+                elif "failing-test-suspects-recipients" == recipient:
+                    if recipient_providers is None:
+                        recipient_providers = XML.SubElement(
+                            email, "recipientProviders"
+                        )
+                    XML.SubElement(
+                        recipient_providers,
+                        "hudson.plugins.emailext.plugins.recipients.FailingTestSuspectsRecipientProvider",
+                    ).text = ""
+                elif "first-failing-build-suspects-recipients" == recipient:
+                    if recipient_providers is None:
+                        recipient_providers = XML.SubElement(
+                            email, "recipientProviders"
+                        )
+                    XML.SubElement(
+                        recipient_providers,
+                        "hudson.plugins.emailext.plugins.recipients.FirstFailingBuildSuspectsRecipientProvider",
+                    ).text = ""
+
+            # `failureCount` is deprecated and has no effect
+            # on email, but the element has been created
+            # in order to match the XML generated via UI.
+            failure_count_supporters = ["FirstFailureTrigger", "SecondFailureTrigger"]
+            if ttype in failure_count_supporters:
+                XML.SubElement(trigger, "failureCount").text = "0"
     else:
-        XML.SubElement(email, "sendToRequester").text = "false"
-        XML.SubElement(email, "sendToDevelopers").text = "false"
-        XML.SubElement(email, "includeCulprits").text = "false"
-        XML.SubElement(email, "sendToRecipientList").text = "true"
+        if plugin_version < pkg_resources.parse_version("2.39"):
```

Comment:
Don't you need to handle the else case? I.e. no 'send-to' in `data` and version >= 2.39?

According to JJB docs recipients is enabled in 'send-to' by default, so I'm curious if this should be handled somehow.

Reply:
It's already handled above. The naming conventions have changed after the version

Reply:
But the above code block is only executed if `send-to` key is in `data`.

I think when `send-to` IS NOT in `data` and version >= 2.39 then JJB defaults are not honored (i.e. `ListRecipientProvider` should be added to XML as `recipients` is marked as true/enabled by default).

What do you think?